contracts/pcv/compound/CompoundBadDebtSentinel.sol

```diff
@@ -0,0 +1,257 @@
+    function isCompoundPcvDeposit(
+        address pcvDeposit
+    ) public view returns (bool) {
+        return compoundPcvDeposits.contains(pcvDeposit);
+    }
+
+    /// @notice getter method to view all current Compound PCV Deposits
+    /// @return all current Compound PCV Deposits
+    function allPcvDeposits() public view returns (address[] memory) {
+        return compoundPcvDeposits.values();
+    }
+
+    /// @notice returns true if the addresses are ordered from least to greatest and contain no duplicates
+    /// @param addresses to check
+    /// @return true if array contains no duplicates and the address are ordered
+    /// returns false if the array has duplicates or is incorrectly ordered.
+    function noDuplicatesAndOrdered(
+        address[] memory addresses
+    ) public pure returns (bool) {
+        /// addresses
+        unchecked {
+            uint256 addressesLength = addresses.length;
+
+            for (uint256 i = 0; i < addressesLength; i++) {
+                if (i + 1 <= addressesLength - 1) {
+                    if (addresses[i] >= addresses[i + 1]) {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+    }
+
+    /// @notice get the total bad debt for a given set of addresses
+    /// @param addresses of users to find sum of bad debt
+    /// @return totalBadDebt of all supplied users
+    function getTotalBadDebt(
+        address[] memory addresses
+    ) public view returns (uint256 totalBadDebt) {
+        uint256 accountsLength = addresses.length;
+
+        for (uint256 i = 0; i < accountsLength; i++) {
+            (, , uint256 badDebt) = IComptroller(comptroller)
+                .getAccountLiquidity(addresses[i]);
+            totalBadDebt += badDebt;
+        }
+    }
+
+    /// ------------- Public State Changing API -------------
+
+    /// @notice rescue funds from all stored PCV Deposits
+    /// no need for the reentrancy lock as the PCV Guardian will lock when withdrawals process
+    /// @param addresses of compound users to query for bad debt
+    function rescueAllFromCompound(
+        address[] memory addresses
+    ) external override {
+        _countBadDebtAndWithdraw(addresses, allPcvDeposits());
+    }
+
+    /// @notice rescue assets from compound, then pause all deposits
+    /// no need for the reentrancy lock as the PCV Guardian will lock when withdrawals process
+    /// @param addresses of compound users to query for bad debt
+    /// @param pcvDeposits to pull funds from
+    function rescueFromCompound(
+        address[] memory addresses,
+        address[] memory pcvDeposits
+    ) external override {
+        uint256 pcvDepositsLength = pcvDeposits.length;
+
+        /// validate pcv deposits are ordered and not duplicated
+        require(
+            noDuplicatesAndOrdered(pcvDeposits),
+            "CompoundBadDebtSentinel: PCVDeposits, Duplicates or OOO"
+        );
```

Comment:
`_countBadDebtAndWithdraw` only checks the ordering of the passed compound user addresses, not the pcv deposits.

`rescueFromCompound` allows the caller to specify the pcv deposits, so we must ensure things are ordered properly and withdraw doesn't get called multiple times on the same pcv deposit

Reply:
However, `rescueAllFromCompound` does not need to ensure ordering and lack of duplicates in the PCV deposits because the user does not specify the deposits to withdraw from in that function.

Reply:
Got it, then you can remove the `noDuplicatesAndOrdered` check in `function _countBadDebtAndWithdraw(...) private`

Reply:
you always have to check that the compound user addresses do not contain duplicates and are ordered. we could remove the `noDuplicatesAndOrdered` call from the `_countBadDebtAndWithdraw` function, but that wouldn't make sense because then we'd have to duplicate that logic of checking for duplicate compound user addresses in both functions `rescueFromCompound` and `rescueAllFromCompound`.

it should be fine as is unless I'm really missing something.

Reply:
we can talk this over on the call later today